core/functions.py

```diff
@@ -36,46 +37,87 @@ def init(path=None, folder=None, config_settings:Dict=None, quiet=False, use_exi
     Then a |JobManager| instance is created as ``config.default_jobmanager`` using *path* and *folder* to determine the main working folder. Settings for this instance are taken from ``config.jobmanager``. If *path* is not supplied, the current directory is used. If *folder* is not supplied, ``plams_workdir`` is used.  If *use_existing_folder* is True and the working folder already exists, PLAMS will not create a new working folder with an incremental suffix (e.g. plams_workdir.002). Instead, it will just use the pre-existing folder (note: that this might lead to issues if the working folder is not empty).
 
     Optionally, an additional `dict` (or |Settings| instance) can be provided to the `config_settings` argument which will be used to update the values from the ``plams_defaults``.
-    
+
+    |init| can be either used as a standalone function or in conjunction with the ``with`` statement, the latter option automatically calling |finish| upon exiting the context manager:
+
+    .. code-block:: python
+
+        >>> from scm.plams import Molecule, Settings, AMSJob, init, finish
+
+        >>> mol: Molecule = ...
+        >>> settings: Settings = ...
+
+        >>> with init():
+        ...     job1 = AMSJob(molecule=mol, settings=settings)
+        ...     result1 = job1.run()
+
+        # Equivalently:
+        >>> init()
+        >>> job2 = AMSJob(molecule=mol, settings=settings)
+        >>> result2 = job2.run()
+        >>> finish()
+
+
     .. warning::
       This function **must** be called before any other PLAMS command can be executed. Trying to do anything without it results in a crash. See also |master-script|.
     """
 
-    if config.init:
-        return
+    __slots__ = ("otherJM",)
```

Comment:
I understand the purpose of __slots__, but since it is not used anywhere else in PLAMS I think it is a little bit confusing and I'm not sure we would be saving significant amount of memory or increasing performance this way

Reply:
I'd say it's still a good practice to be explicit with which attributes are defined for a class (which `__slots__` absolutely helps with), but it's your call. Removed as of now.